Module 5/Problem_1/Q1.json

```diff
@@ -0,0 +1,182 @@
+          "state": "CA",
+          "zip": "12345"
+        }
+      },
+      "items": [
+        {
+          "itemId": 1,
+          "name": "Product A",
+          "quantity": 2,
+          "price": 10
+        },
+        {
+          "itemId": 2,
+          "name": "Product B",
+          "quantity": 1,
+          "price": 20
+        }
+      ]
+    },
+    {
+      "orderId": 2,
+      "orderDate": "2022-02-21T09:12:34Z",
+      "customer": {
+        "customerId": 2,
+        "firstName": "Jane",
+        "lastName": "Smith",
+        "email": "janesmith@example.com",
+        "address": {
+          "street": "456 Oak St",
+          "city": "Somewhere",
+          "state": "NY",
+          "zip": "67890"
+        }
+      },
+      "items": [
+        {
+          "itemId": 3,
+          "name": "Product C",
+          "quantity": 3,
+          "price": 15
+        }
+      ]
+    }
+  ]
+}
+
+json Spec:
+[
+  {
+    "operation": "modify-overwrite-beta",
+    "spec": {
+      "orders": {
+        "*": {
+          "customer": {
+            "address": "=concat(@(1,address.street),' ',@(1,address.city), ' ', @(1,address.state), ' ',@(1,address.zip))",
+            "fullname": "=concat(@(1,firstName),' ',@(1,lastName))"
+          },
+          "itemTemp": "@(1,items)"
+        }
+      }
+    }
+  },
+
+  {
+    "operation": "modify-overwrite-beta",
+    "spec": {
+      "orders": {
+        "*": {
+          "itemTemp": {
+            "*": {
+              "inverseValue": "=divide(1,@(1,price))",
+              "temp": "=divideAndRound(2, @(1,quantity), @(1,inverseValue))"
+            }
+          }
+        }
+      }
+    }
+  },
+  {
+    "operation": "shift",
+    "spec": {
+      "orders": {
+        "*": {
+          "*": "orders[&1].&",
+          "itemTemp": {
+            "*": {
+              "temp": "orders[#4].total"
+            }
+          }
+        }
+      }
+    }
+    },
+  {
+    "operation": "modify-overwrite-beta",
+    "spec": {
+      "orders": {
+        "*": {
+          "total": "=doubleSum(@(1,total))"
+        }
+      }
+    }
+    },
+
+  {
+    "operation": "remove",
+    "spec": {
+      "orders": {
+        "*": {
+          "items": {
+            "*": {
+              "inverseValue": "",
+              "temp": ""
+            }
+          },
+          "customer": {
+            "firstName": "",
+            "lastName": ""
+          }
+        }
+      }
+    }
+    }
+]
+
+Output:
+{
+  "orders" : [ {
+    "orderId" : 1,
+    "orderDate" : "2022-02-20T12:34:56Z",
+    "customer" : {
+      "customerId" : 1,
+      "firstName" : "John",
+      "lastName" : "Doe",
+      "email" : "johndoe@example.com",
+      "address" : "123 Main St Anytown CA 12345"
+    },
+    "items" : [ {
+      "itemId" : 1,
+      "name" : "Product A",
+      "quantity" : 2,
+      "price" : 10
+    }, {
+      "itemId" : 2,
+      "name" : "Product B",
+      "quantity" : 1,
+      "price" : 20
```

Comment:
Price should be in decimal

Reply:
Converted it into double using the 'toDouble' function.